pulser/tests/test_simresults.py

```diff
@@ -109,33 +134,53 @@ def test_expect():
     assert len(results.expect(op)[0]) == number_of_meas
 
 
+def test_expect_noisy():
+    bad_op = qutip.tensor([qutip.qeye(2), qutip.sigmap()])
+    with pytest.raises(ValueError, match="non-diagonal"):
+        results_noisy.expect([bad_op])
+    op = qutip.tensor([qutip.qeye(2), qutip.basis(2, 0).proj()])
+    results_noisy.expect([op])
```

Comment:
Same comment here: do some assertation, check that you get what you expect when things go right

Reply:
Can you assert that `results_noisy` is what you expect?